numpy/polynomial/polynomial.py

```diff
@@ -1252,13 +1260,26 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None):
         diagnostic information from the singular value decomposition (used
         to solve the fit's matrix equation) is also returned.
     w : array_like, shape (`M`,), optional
-        Weights. If not None, the weight ``w[i]`` applies to the unsquared
-        residual ``y[i] - y_hat[i]`` at ``x[i]``. Ideally the weights are
-        chosen so that the errors of the products ``w[i]*y[i]`` all have the
-        same variance.  When using inverse-variance weighting, use
-        ``w[i] = 1/sigma(y[i])``.  The default value is None.
-
+        Weights to apply to the unsquared residual of each point.  When 
+        these correspond to weighting by Gaussian uncertainties,
+        use w = 1/sigma.  
+        Note that w here is the sqrt of the more typical definition 
+        of the weights in weighted least squares, which generally 
+        uses W = w**2 = 1/sigma**2 (or more generally 1/var) to weight the 
+        squared residual of each point.
         .. versionadded:: 1.5.0
+    cov : bool, optional
+        If True, return the covariance matrix.
+        .. versionadded:: 1.25.2
+    absolute_w: bool, optional
+        If False, the covariance are scaled by chi2/dof, where 
```

Comment:
Getting nitpicky here.. I don't know if it's 100% accurate to say `chi2/dof`. You might want to say `residuals/dof` or `cost/dof`. But this might be my lack of complete understanding on the terminology. The usage of "reduced chi2" below seems correct to me. It's just this usage of `chi2` above that I'm worried about. That said, I think it's likely all readers (who know the statistics involved here) would understand what you mean be `chi2/dof`. But they key point is what you say below: "everything is scaled such that the reduced chi2 ... is unity".